pkg/venafi/tpp/connector.go

```diff
@@ -758,6 +786,43 @@ func (c *Connector) RequestCertificate(req *certificate.Request) (requestID stri
 	return
 }
 
+// For the purpose of using this function in RequestCertificate, we ignore the
+// errors "certificate not found" and "reset was not needed".
+func (c *Connector) reset(certificateDN string, restart bool) error {
+	statusCode, status, body, err := c.request("POST", urlResourceCertificateReset, certificateResetRequest{
+		CertificateDN: certificateDN,
+		Restart:       restart,
+	})
+	if err != nil {
+		return fmt.Errorf("while resetting certificate due to a past failed enrollment: %w", err)
+	}
+
+	switch {
+	case statusCode == http.StatusOK:
+		return nil
+	case statusCode == http.StatusBadRequest:
+		var decodedResetResponse certificateRequestResponse
+		if err := json.Unmarshal(body, &decodedResetResponse); err != nil {
+			return fmt.Errorf("failed to decode reset response: HTTP %d: %s: %s", statusCode, status, body)
+		}
+
+		// We don't need to error out if the certificate is already reset.
+		if decodedResetResponse.Error == "Reset is not completed. No reset is required for the certificate." {
+			return nil
+		}
+
+		// This is specific to RequestCertificate: we want to silently skip
+		// certificates that don't exist.
+		if strings.HasSuffix(decodedResetResponse.Error, "does not exist or you do not have sufficient rights to the object.") {
+			return nil
+		}
+
+		return fmt.Errorf("while resetting certificate due to a past failed enrollment. Status: %s, Body: ", status, string(body))
+	default:
+		return fmt.Errorf("while resetting certificate due to a past failed enrollment. Status: %s, Body: ", status, string(body))
```

Comment:
can we combine these two return statements?

Reply:
There was no reason to dump the body in the first `return`, so I changed to this:

```go
		return fmt.Errorf("while resetting certificate due to a past failed enrollment: %s", decodedResetResponse.Error)
	default:
		return fmt.Errorf("while resetting certificate due to a past failed enrollment. Status: %s, Body: %s", status, string(body))
```

The two statements are now different, should I still combine them?

Reply:
No,  thank you for fixing!